hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DistributionSetManagementTest.java

```diff
@@ -818,6 +819,28 @@ public void deleteAssignedDistributionSet() {
                         .isEqualTo(2);
     }
 
+    @Test
+    @Description("Test different assignment results, after assign a ds")
+    public void testAlreadyAssignedTargetsToDs() {
+        DistributionSet dsToTargetAssigned = testdataFactory.createDistributionSet("ds-3");
+
+        // create assigned DS
+        dsToTargetAssigned = distributionSetManagement.findDistributionSetByNameAndVersion(dsToTargetAssigned.getName(),
+                dsToTargetAssigned.getVersion());
+        final Target target = new JpaTarget("4712");
+        final Target savedTarget = targetManagement.createTarget(target);
+        final List<Target> toAssign = new ArrayList<>();
+        toAssign.add(savedTarget);
+        DistributionSetAssignmentResult assignmentResult = deploymentManagement
+                .assignDistributionSet(dsToTargetAssigned, toAssign);
+        assertThat(assignmentResult.getAssignedEntity()).hasSize(1);
+
+        assignmentResult = deploymentManagement.assignDistributionSet(dsToTargetAssigned, toAssign);
+        assertThat(assignmentResult.getAssignedEntity()).hasSize(0);
```

Comment:
But this is not the test. This test should verfiy that  the `DistributionSetAssignmentResult`  not contains already assigned targets.